File: js-automation/pages/CartPage.js
import { By } from 'selenium-webdriver';
import { BasePage } from './BasePage.js';

export class CartPage extends BasePage {
  constructor(driver, baseUrl) {
    super(driver, baseUrl);
    this.cartItems = By.css("[data-testid='cart-item']");
    this.emptyCartMessage = By.css("[data-testid='cart-empty']");
    this.checkoutButton = By.css("[data-testid='checkout-button'], #cart-checkout-button");
    this.grandTotal = By.css("[data-testid='cart-grand-total']");
    this.removeButton = By.css("[data-testid='remove-from-cart']");
    this.quantityIncrease = By.css("[data-testid='quantity-increase']");
    this.quantityDecrease = By.css("[data-testid='quantity-decrease']");
    this.itemQuantity = By.css("[data-testid='cart-item-quantity']");
  }

  async load() {
    await this.navigateTo('/cart');
  }

  async getItemCount() {
    const items = await this.driver.findElements(this.cartItems);
    return items.length;
  }

  async proceedToCheckout() {
    await this.click(this.checkoutButton);
    await this.sleep(500);
  }

  async removeItem() {
    await this.click(this.removeButton);
    await this.sleep(500);
  }

  async increaseFirstItemQuantity() {
    await this.click(this.quantityIncrease);
    await this.sleep(400);
  }

  async getFirstItemQuantity() {
    return await this.getText(this.itemQuantity);
  }

  async getGrandTotal() {
    return await this.getText(this.grandTotal);
  }
}

File: automation/Pages/CartPage.cs
using System;
using System.Linq;
using OpenQA.Selenium;
using NOVAMart.Automation.Configuration;
using NOVAMart.Automation.Utilities;

namespace NOVAMart.Automation.Pages
{
    public class CartPage : BasePage
    {
        public CartPage(IWebDriver driver) : base(driver) { }

        public IWebElement CheckoutButton =>
            WaitHelper.WaitForElementClickable(Driver, By.CssSelector("[data-testid='checkout-button']"));

        public void NavigateTo()
        {
            Driver.Navigate().GoToUrl($"{TestConfiguration.Instance.BaseUrl}/cart");
            WaitHelper.WaitForElementVisible(Driver, By.TagName("body"));
        }

        public int GetCartItemCount()
        {
            try
            {
                return Driver.FindElements(By.CssSelector("[data-testid='cart-item']")).Count;
            }
            catch
            {
                return 0;
            }
        }

        public int GetFirstItemQuantity()
        {
            var elem = Driver.FindElements(By.CssSelector("[data-testid='cart-item-quantity']")).FirstOrDefault();
            if (elem != null && int.TryParse(elem.Text.Trim(), out var qty))
            {
                return qty;
            }
            return 0;
        }

        public void IncreaseFirstItemQuantity()
        {
            var btn = WaitHelper.WaitForElementClickable(Driver, By.CssSelector("[data-testid='quantity-increase']"));
            btn.Click();
            System.Threading.Thread.Sleep(500);
        }

        public void DecreaseFirstItemQuantity()
        {
            var btn = WaitHelper.WaitForElementClickable(Driver, By.CssSelector("[data-testid='quantity-decrease']"));
            btn.Click();
            System.Threading.Thread.Sleep(500);
        }

        public void RemoveFirstItem()
        {
            var btn = WaitHelper.WaitForElementClickable(Driver, By.CssSelector("[data-testid='remove-from-cart']"));
            btn.Click();
            System.Threading.Thread.Sleep(500);
        }

        public void ProceedToCheckout()
        {
            CheckoutButton.Click();
            WaitHelper.WaitForUrlContains(Driver, "/checkout");
        }

        public bool IsEmptyCartDisplayed()
        {
            try
            {
                var elem = WaitHelper.WaitForElementVisible(Driver, By.CssSelector("[data-testid='cart-empty']"), 4);
                return elem != null && elem.Displayed;
            }
            catch
            {
                return false;
            }
        }

        public string GetGrandTotal()
        {
            try
            {
                var totalElem = Driver.FindElement(By.CssSelector("[data-testid='cart-grand-total']"));
                return totalElem.Text.Trim();
            }
            catch
            {
                return string.Empty;
            }
        }
    }
}
